listener.go

```diff
@@ -67,6 +67,17 @@ func (l *listener) Accept() (net.Conn, error) {
 	return Server(c, l.config)
 }
 
+// AcceptNoHandshake returns a connection that has not completed handshake
```

Comment:
I'd be tempted to create a new type of "no handshake" listener that still implements `net.Listener`. This would maintain consistent use of a familiar interface and also makes maintaining this fork a bit easier as we've only added code instead of modifying.  Something along the lines of:

```golang
// NewNoHandshakeListener creates a DTLS listener which accepts connections from
// an inner Listener but does not automatically handshake on Accept.
func NewNoHandshakeListener(inner net.Listener, config *Config) (net.Listener, error) {
	if err := validateConfig(config); err != nil {
		return nil, err
	}

	return &noHandshakeListener{
		config: config,
		parent: inner,
	}, nil
}

// noHandshakeListener represents a DTLS listener that does not automatically
// handshake on Accept.
type noHandshakeListener struct {
	config *Config
	parent net.Listener
}

// Accept waits for and returns the next connection to the listener.
// Connection handshake does not occur during accept and needs to be manually
// called by passed the returned net.Conn to the Server function.
func (n *noHandshakeListener) Accept() (net.Conn, error) {
	return n.parent.Accept()
}

// Close closes the listener.
// Any blocked Accept operations will be unblocked and return errors.
// Already Accepted connections are not closed.
func (n *noHandshakeListener) Close() error {
	return n.parent.Close()
}

// Addr returns the listener's network address.
func (n *noHandshakeListener) Addr() net.Addr {
	return n.parent.Addr()
}
```

Reply:
that's a great idea, will do!

Reply:
Did something in that spirit in e78f28d